src/main.rs

```diff
@@ -119,3 +132,17 @@ fn main() {
     }
 
 }
+
+fn discover_codeowners() -> Option<PathBuf> {
+    let curr_dir = current_dir().unwrap();
+    let repo = match Repository::discover(&curr_dir) {
```

Comment:
I wonder if we should just use std::process::Command with "git rev-parse --show-toplevel" since there's not much heavy git lifting needed here. 

How much bigger did adding a crate to perform this a action add?

Reply:
Here are some numbers I get:

**master branch**
```sh
# Compilation time
$ rm -rf target/ && time cargo build --release
...
    Finished release [optimized] target(s) in 1m 12s

real	1m12.546s
user	3m13.753s
sys	0m3.983s

# Binary size
$ du -h target/release/git-codeowners
1.7M	target/release/git-codeowners
```
**PR branch**
```sh
$ rm -rf target/ && time cargo build --release
...
    Finished release [optimized] target(s) in 1m 58s

real	1m49.198s
user	6m23.143s
sys	0m19.478s

$ du -h target/release/git-codeowners
2.0M	target/release/git-codeowners
```
Thanks for encouraging me to measure this. I expect that whatever performance benefit to be had from leveraging git2 is likely  _not_ worth doubling the userland CPU time required to compile. But I'll try using std::process::Command as you described and see what perf difference I can measure.